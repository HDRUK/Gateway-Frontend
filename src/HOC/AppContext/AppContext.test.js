import React from "react";
import { create } from "react-test-renderer";
import AppContextProvider from "./AppContext";

describe("<AppContextProvider> ", () => {
    let renderedComponent;
    let componentInstance;

    beforeEach(() => {
        renderedComponent = create(
            <AppContextProvider>
                <div>test</div>
            </AppContextProvider>
        );
        componentInstance = renderedComponent.getInstance();
    });

    it("should test the AppContextProvider HOC", () => {
        expect(componentInstance.state.counter).toEqual(0);
        expect(componentInstance.counterFunc).toBeDefined();
        expect(renderedComponent).toMatchSnapshot();
    });

    describe("when counterFunc is called", () => {
        componentInstance.counterFunc();

        it("should increase state counter by 1", () => {
            expect(componentInstance.state.counter).toEqual(1);
        });
    });
});
