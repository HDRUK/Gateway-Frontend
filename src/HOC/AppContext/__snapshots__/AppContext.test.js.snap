// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContextProvider>  should test the AppContextProvider HOC 1`] = `
<div>
  test
</div>
`;
